### PR TITLE
feat: support name only replacements

### LINKED ISSUE #14386: Replacements are filtered out if the version is exactly the same
### How are you running Renovate?

WhiteSource Renovate hosted app on github.com

### If you're self-hosting Renovate, tell us what version of Renovate you run.

_No response_

### Please select which platform you are using if self-hosting.

_No response_

### If you're self-hosting Renovate, tell us what version of the platform you run.

_No response_

### Was this something which used to work for you, and then stopped?

I never saw this working

### Describe the bug

Seen in the replacement for `@emotion/core` to `@emotion/react`. The replacement as defined in https://github.com/renovatebot/renovate/pull/14367 doesn't work when the old version and the replacement version are identical.

The replacement is added here:

https://github.com/renovatebot/renovate/blob/368903ab814f5b19117eaab6c2cb33382be6cad6/lib/workers/repository/process/lookup/index.ts#L150-L160

And then filtered out here:

https://github.com/renovatebot/renovate/blob/368903ab814f5b19117eaab6c2cb33382be6cad6/lib/workers/repository/process/lookup/index.ts#L360-L367

We need to add another condition for `update.updateType === 'replacement'`

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste any log here, between the starting and ending backticks
```

</details>

### Have you created a minimal reproduction repository?

No reproduction repository

### LINKED ISSUE #20628: Support name-only replacements
### What would you like Renovate to be able to do?

Add support for `replacementName` package rules which don't require a `replacementVersion` to be specified.

Use-cases are for migrating from `java` or `adoptopenjdk` -> `eclipse-temurin` without changing the current version value.

### If you have any ideas on how this should be implemented, please tell us here.

Add new condition to lookup/index.ts

### Is this a feature you are interested in implementing yourself?

Yes

### PR DESCRIPTION
**Pull request by setchy**
## Changes

Supports replacement packageRules that only specify a replacementName.  This is useful for situations where you wish to maintain existing versions but update the dependency name (eg: `java` or `adoptopenjdk` -> `eclipse-temurin`)

## Context

Closes https://github.com/renovatebot/renovate/issues/20628 
Closes https://github.com/renovatebot/renovate/issues/14386

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/workers/repository/update/branch/auto-replace.ts
Comment (setchy) [inline lib/workers/repository/update/branch/auto-replace.ts:222]: This conditional check was causing a lot of heartache with this change.  the confirmIfDepUpdated check is run 3 other times during the process, and i couldn't see what value it was adding here.  Keen on feedback from the maintainers on this change.  

I did spend a fair amount of time trying to leave this check in-place and tweak the logic inside confirmIfDepUpdated, but that uncovered even more side-effects

File: lib/workers/repository/process/lookup/index.spec.ts
Comment (viceice) [inline lib/workers/repository/process/lookup/index.spec.ts:1852]: please use `toEqual`

File: lib/workers/repository/update/branch/auto-replace.ts
Comment (viceice) [reply lib/workers/repository/update/branch/auto-replace.ts:222]: i think this is for cases where multiple deps are using a shared variable. 🤔

Comment (setchy) [reply lib/workers/repository/update/branch/auto-replace.ts:222]: Will await https://github.com/renovatebot/renovate/pull/20627/files being resolved - since i now think this guard can be added back after the `confirmIfDepUpdated` refactor

**Comment by viceice** [comment]
@setchy can this now continued?

**Comment by setchy** [comment]
> @setchy can this now continued?

Indeed. Will work on the unit test changes today/tomorrow

File: lib/workers/repository/process/lookup/index.ts
Comment (viceice) [inline lib/workers/repository/process/lookup/index.ts:162]: what happens when we've a docker image without tag and digest. `currentValue` can probably be undefined in that case.

Comment (setchy) [reply lib/workers/repository/process/lookup/index.ts:162]: Just did a quick test and that scenario at the moment isn't handled 😅

Comment (setchy) [reply lib/workers/repository/process/lookup/index.ts:162]: updated to handle this scenario.  testing on two real repositories also

File: lib/workers/repository/process/lookup/index.spec.ts
Comment (viceice) [inline lib/workers/repository/process/lookup/index.spec.ts:1854]: can you add a second greater version to be sure we don't break it in future

File: lib/workers/repository/process/lookup/index.spec.ts
Comment (viceice) [inline lib/workers/repository/process/lookup/index.spec.ts:1878]: same here

File: lib/workers/repository/process/lookup/index.spec.ts
Comment (viceice) [inline lib/workers/repository/process/lookup/index.spec.ts:1912]:
```suggestion
          newName: 'eclipse-temurin',
          newValue: undefined,
```
it's still there

File: lib/workers/repository/process/lookup/index.spec.ts
Comment (setchy) [reply lib/workers/repository/process/lookup/index.spec.ts:1887]: test case updated

File: lib/workers/repository/process/lookup/index.spec.ts
Comment (setchy) [reply lib/workers/repository/process/lookup/index.spec.ts:1854]: test case updated

File: lib/workers/repository/process/lookup/index.spec.ts
Comment (setchy) [reply lib/workers/repository/process/lookup/index.spec.ts:1928]: added

**Comment by setchy** [comment]
Well that's annoying - individually both tests pass, but as a suite they fail 🤦

File: lib/workers/repository/process/lookup/index.spec.ts
Comment (viceice) [inline lib/workers/repository/process/lookup/index.spec.ts:1908]:
```suggestion
```

duplicate mock

File: lib/workers/repository/process/lookup/index.spec.ts
Comment (viceice) [inline lib/workers/repository/process/lookup/index.spec.ts:1865]:
```suggestion
```

unused

File: lib/workers/repository/process/lookup/index.spec.ts
Comment (viceice) [inline lib/workers/repository/process/lookup/index.spec.ts:50]:
```suggestion
    // TODO: fix wrong tests
    jest.resetAllMocks();
    // TODO: fix types #7154
```

**Review by viceice — APPROVED** [review]

**Comment by viceice** [comment]
after merge, we should add two rules.
- `containerbase/buildpack` to `containerbase/base`
- `renovatebot/internal-tools` to `containerbase/internal-tools`

**Pull request merged**
